manage-data/data-store/data-streams/failure-store-recipes.md

```diff
@@ -0,0 +1,1153 @@
+
+:::{image} /manage-data/images/elasticsearch-reference-management_failure_store_alerting_esql.png
+:alt: use the ::failures suffix in the data stream name in the from command
+:::
+::::
+
+::::{step} Test
+Configure schedule, actions, and details of the alert before saving the rule.
+
+:::{image} /manage-data/images/elasticsearch-reference-management_failure_store_alerting_finish.png
+:alt: complete the rule configuration and save it
+:::
+::::
+
+::::{step} Done
+::::
+
+:::::
+
+## Data remediation [failure-store-recipes-remediation]
+
+If you've encountered a long span of ingestion failures you may find that a sizeable gap of events has appeared in your data stream. If the failure store is enabled, the documents that should fill those gaps would be tucked away in the data stream's failure store. Because failure stores are made up of regular indices and the failure documents contain the document source that failed, the failure documents can often times be replayed into your production data streams.
+
+::::{warning}
+Care should be taken when replaying data into a data stream from a failure store. Any failures during the replay process may generate new failures in the failure store which can duplicate and obscure the original events.
+::::
+
+We recommend a few best practices for remediating failure data.
+
+**Separate your failures beforehand.** As described in the [failure document source](#use-failure-store-document-source) section above, failure documents are structured differently depending on when the document failed during ingestion. We recommend to separate documents by ingest pipeline failures and indexing failures at minimum. Ingest pipeline failures often need to have the original pipeline re-executed, while index failures should skip any pipelines. Further separating failures by index or specific failure type may also be beneficial.
+
+**Perform a failure store rollover.** Consider rolling over the failure store before attempting to remediate failures. This will create a new failure index that will collect any new failures during the remediation process.
+
+**Use an ingest pipeline to convert failure documents back into their original document.** Failure documents store failure information along with the document that failed ingestion. The first step for remediating documents should be to use an ingest pipeline to extract the original source from the failure document and discard any other info on it.
+
+**Simulate first to avoid repeat failures.** If you must execute a pipeline as part of your remediation process, it is best to simulate the pipeline against the failure first. This will catch any unforeseen issues that may fail the document a second time. Remember, ingest pipeline failures will capture the document before an ingest pipeline was applied to it, which can further complicate remediation when a failure document becomes nested inside a new failure.
```

Comment:
```suggestion
**Simulate first to avoid repeat failures.** If you must run a pipeline as part of your remediation process, it is best to simulate the pipeline against the failure first. This will catch any unforeseen issues that may fail the document a second time. Remember, ingest pipeline failures will capture the document before an ingest pipeline is applied to it, which can further complicate remediation when a failure document becomes nested inside a new failure.
```
rather than `will` capture and `was` applied, maybe we use future and present tense here.